Assets/MRTK/Core/Providers/BaseGlobalInputSource.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Linq;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Base class for input sources whose pointers are all active pointers in the scene.
+    /// </summary>
+    /// <remarks>
+    /// <para>This base class is intended to represent input sources which raise event to all active pointers found by the FocusProvider in a scene.</para>
+    /// </remarks>
+    public class BaseGlobalInputSource : IMixedRealityInputSource, IDisposable
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        public BaseGlobalInputSource(string name, IMixedRealityFocusProvider focusProvider, InputSourceType sourceType = InputSourceType.Other)
+        {
+            SourceId = (CoreServices.InputSystem != null) ? CoreServices.InputSystem.GenerateNewSourceId() : 0;
+            SourceName = name;
+            FocusProvider = focusProvider;
+            SourceType = sourceType;
+        }
+
+        /// <inheritdoc />
+        public uint SourceId { get; }
+
+        /// <inheritdoc />
+        public string SourceName { get; }
+
+        /// <inheritdoc />
+        public virtual IMixedRealityPointer[] Pointers { get; set; }
+
+        /// <inheritdoc />
+        public InputSourceType SourceType { get; set; }
+
+        private IMixedRealityFocusProvider FocusProvider;
+
+        public void UpdateActivePointers()
+        {
+            Pointers = FocusProvider.GetPointers<IMixedRealityPointer>().Where(x => x.IsActive).ToArray();
```

Comment:
Do you think we need to filter out near pointers vs far or something? Or do you think any should be valid?

Reply:
I think all should be valid, the idea of this kind of input source is that it will send events to all active pointers, not just pointers of a specific type (though we could also make an input source for that case!)